pkg/podimpersonation/podimpersonation.go

```diff
@@ -559,9 +568,33 @@ func (s *PodImpersonation) augmentPod(pod *v1.Pod, sa *v1.ServiceAccount, secret
 				continue
 			}
 
+			vmount := v1.VolumeMount{
+				Name:      "user-kubeconfig",
+				MountPath: "/tmp/.kube",
+			}
+			cfgVMount := v1.VolumeMount{
+				Name:      "user-kube-configmap",
+				MountPath: "/home/.kube/config",
+				SubPath:   "config",
+			}
+
+			pod.Spec.InitContainers = []v1.Container{
+				{
```

Comment:
Due to the new `init-container` when I click on the `Kubectl shell` button on Rancher Manager, it is taking 18 seconds to load the shell.

Before the `init-container` it takes only 4 seconds to load the shell.

This is a serious degradation of user experience.

Maybe due to this, I find the `sed` solution better, if we can't find a way to mitigate this user experience issue.

Reply:
It was the `imagePullPolicy` that was causing this. The default value is `Always` and I changed it to `PullIfNotPresent`. Should be fine now